### ISSUE TITLE
[false-negative]IntLongMath can not detect the case int converted to long when passed as a function parameter.

### ISSUE THREAD

**Issue by wuchiuwong** [google/error-prone#1662]
### Description of the problem / feature request:

IntLongMath can not detect the case int converted to long when passed as a function parameter.
This rule is implemented in error_prone_core-2.3.4-sources.jar!\com\google\errorprone\bugpatterns\IntLongMath.java，
When searching for errors, this rule detects whether return statements(<code>matchReturn</code>), variable assignment statements(<code>matchAssignment</code>), and variable definition statements(<code>matchVariable</code>) contain int convert into long,
Therefore, the type conversion that occurs as a function parameter is ignored.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```
Thread.sleep(updateInterval + (new Random().nextInt(300) * 1000));
file.seek(sectorNumber * SECTOR_BYTES);
```

### What version of Error Prone are you using?

2.3.4

###  Have you found anything relevant by searching the web?

no

**Comment by graememorgan** [comment]
Are you suggesting matching every situation where an `int` is widened to `long` automatically, or is there an additional heuristic you're looking for?
